### PR TITLE
Add gather_point_op  and query_ball_op

### PR DESCRIPTION
**Pull request by tink2123**
Add two op for PointNet++
**1. gather_point**  : https://github.com/erikwijmans/Pointnet2_PyTorch/blob/49501a0b0d3a5811c6cc2baa5b1a2144d5390c56/pointnet2/_ext-src/src/sampling_gpu.cu
**2. query_ball**: https://github.com/erikwijmans/Pointnet2_PyTorch/blob/49501a0b0d3a5811c6cc2baa5b1a2144d5390c56/pointnet2/_ext-src/src/ball_query_gpu.cu

### CODE REVIEW

File: paddle/fluid/operators/gather_point_op.cc [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                   "Index of GatherPointop should be 2-D Tensor");
+    ctx->SetOutputDim("Output", {x_dims[0], index_dims[1], 3});
+  }
+
+ protected:
+  framework::OpKernelType GetExpectedKernelType(
+      const framework::ExecutionContext &ctx) const override {
+    auto input_data_type = ctx.Input<Tensor>("X")->type();
+    return framework::OpKernelType(input_data_type, ctx.device_context());
+    // return framework::OpKernelType(input_data_type, ctx.GetPlace());
```
Comment (heavengate) [inline paddle/fluid/operators/gather_point_op.cc]: 这里应该是ctx.GetPlace() ？

Comment (tink2123) [reply paddle/fluid/operators/gather_point_op.cc]: > 这里应该是ctx.GetPlace() ？
是的，已修改～thanks！

**Review by heavengate — APPROVED** [review]
LGTM

**Review by heavengate — APPROVED** [review]
LGTM

**Pull request merged**